docs/roadmap.md

```diff
@@ -0,0 +1,170 @@
+
+## Simple installation
+
+The first you do with an IDE is boring but critical: import a project. The
+Language Server Protocol does not provide utilities to extract metadata from a
+build tool (module source files, dependencies, configuration data), so we are
+still exploring what is the best option.
+
+* [ ] sbt-metals: plugin that provides a `*:metalsSetup` task to export sbt
+      build metadata.
+* [ ] [BSP][]: LSP-inspired protocol to standardize on communication between a
+      language server and build tool. In theory, should enable automatic
+      importing of projects without custom build plugins or manual installation
+      steps.
+
+## Correct diagnostics
+
+Red squiggles are one of the most basic but also difficult-to-support features
+in an IDE. The Scala compiler was not designed for interactivity, which makes
+fast and correct diagnostics difficult to support in the editor. We are still
+unsure what the best model is for diagnostics, so we are still exploring what is
+the best option.
+
+* [x] Scala Presentation Compiler: interactively publish squiggles as you type.
+      Provides low latency but may get off-sync with build diagnostics.
+  * Disabled by default. Can be enabled with configuration option
+    `scalac.diagnostics.enabled=true`.
+  * Requires [`metalsSetup`](#metalssetup).
+* [x] sbt server: sbt 1.1 added LSP support so that build errors are reported
+      directly to the editor. Metals can connect to sbt server as a client to
+      trigger batch compilation on file save. Guarantees diagnostics are always
+      in-sync with build diagnostics but at the cost of higher latency.
+  * Disabled by default. Can be enabled with configuration option
+    `sbt.enabled=true`.
+  * Requires sbt 1.1 or later.
+* [x] Linting with [Scalafix][].
+  * Requires [semanticdb-scalac](#semanticdb-scalac).
+  * Supports both built-in and custom linter rules.
+
+## Robust navigation
+
+Goto definition and find references is a great way to navigate and learn about a
+codebase. Metals uses [SemanticDB][], a data model for semantic information
+about programs in Scala and other languages, to power code navigation. Indexing
+happens with during batch compilation in the build tool, using a similar
+architecture as [Index-While-Building][] in XCode 9.
+
+* Goto definition (`textDocument/definition`). Requires
+  [semanticdb-scalac](#semanticdb-scalac).
+  * [x] project -> project.
+  * [x] project -> Scala dependency. Requires [`metalsSetup`](#metalssetup).
+  * [x] project -> Java dependency. Requires [`metalsSetup`](#metalssetup).
+  * [ ] dependency -> dependency.
+* [x] Find references (`textDocument/references`). Works the same as goto
+      definition.
+  * [x] project -> project.
+  * [ ] project -> Scala dependency.
+  * [ ] project -> Java dependency.
+  * [ ] dependency -> dependency.
+* [x] Highlight references to symbol at position
+      (`textDocument/documentHighlight`).
```

Comment:
It's currently disabled. Should we enable it by default or mention here that it's an opt-in?

Reply:
I think we should enable it by default. It uses same data as references which is enabled by default.

Reply:
Well, even if it uses the same data, when it's lagging, it's much more noticeable (than jumping to a bit wrong place):

<img width="435" alt="screen shot 2018-01-15 at 14 51 46" src="https://user-images.githubusercontent.com/766656/38212064-c3e17e54-36bc-11e8-9ca3-1ae9e218163c.png">